internal/service/http/http.go

```diff
@@ -243,6 +255,76 @@ func (s *Service) Run(ctx context.Context, host service.Host) error {
 	return nil
 }
 
+func (s *Service) supportHandler(rw http.ResponseWriter, r *http.Request) {
+	s.winMut.Lock()
+	cfg := s.opts
+	s.winMut.Unlock()
+
+	runtime, ok := s.host.(*alloy_runtime.Runtime)
+	if !ok {
+		level.Error(s.log).Log("msg", "failed to get runtime for support bundle", "err", "host is not a runtime")
+	}
```

Comment:
this does not feel so great to cast the runtime from the host interface but extending the host interface with the temporary logger func does not feel so great either. I don't see another easy way though

Reply:
I agree, I looked at refactoring to use the internal logger directly in the http service, I can take another pass at it to see if there's a cleaner way.

Reply:
I did this refactor (changed to using the logging.Logger directly instead of the go-kit interfaces, and there's just one thing I really don't like about it. I didn't want to reimplement log.With in our internal logging structures, so I just created an internalLog function for the http service specifically that would add the "service", "http" context to each log. I don't love this change, but updating the logger implementation to support the log.With pattern was going to be a pretty significant change to the logging implementation that I wanted to avoid.

Let me know what you think!

Reply:
Would it work to pass both the logging.Logger to set the temporary logger + the one with log.With to log inside of the HTTP service?